Add how-to for ClientSettingsPolicy (#2071)
nginxinc/nginx-gateway-fabric: site/content/how-to/monitoring/tracing.md

This guide explains how to enable tracing on HTTPRoutes using NGINX Gateway Fabric. It uses the OpenTelemetry Collector and Jaeger to process and collect the traces.

{{< important >}}

> Tracing cannot be enabled for HTTPRoutes that have [HTTPRouteMatches](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPRouteMatch) with `headers`, `params`, or `method` matchers defined.

Reviewer: Should we also mention that this will be fixed in a future release? (for both docs)

Just want the language to be clear that this is a bug, not by design.

Author: Sure. Something like: "This will be addressed in a future release"?

Author: Or "This is a known issue that will be fixed in a future release".

Reviewer: The first one sounds good.

Reviewer: Should we also say `matches` with those conditions? Because it should work for any paths in the HTTPRoute otherwise that don't have the matches.

Author: > Tracing cannot be enabled for [HTTPRoute matches](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPRouteMatch) with `headers`, `params`, or `method` matchers defined. This will be addressed in a future release.

?

Author: Does that make sense?

Reviewer: Sounds good.

> Tracing cannot be enabled for [HTTPRoute matches](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPRouteMatch) with `headers`, `params`, or `method` matchers defined. This will be addressed in a future release.

Reviewer 2: ```suggestion
Tracing cannot be enabled for [HTTPRoute matches](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPRouteMatch) with `headers`, `params`, or `method` matchers defined. It will be added in a future release.
```
"Addressed" is ambiguous: added has a higher amount of specificity for intent to implement.
